straxen/plugins/s2cnn_processing.py

```diff
@@ -0,0 +1,154 @@
+    It uses doublewidth transformation to get from 
+    HEX placing of PMTs to standard square binning
+    """
+    def PMTnumuber_doublewidth(self):
+        """
+        This function calculates the relation between 
+        PMT channel number and position on 2D "image"
+        """
+        coords = {}
+        row_nPMT = np.array([0, 6, 9, 12, 13, 14, 15, 
+                             16, 17, 16, 17, 16, 17, 16, 
+                             15, 14, 13, 12, 9, 6])
+        row_nPMT_cumulative = np.cumsum(row_nPMT)
+        tot_rows = len(row_nPMT)-2 #
+        pairs = []
+        for i in range(0, 253):
+            n_row = np.argwhere( (i>=row_nPMT_cumulative[0:-1])*
+                                 (i<row_nPMT_cumulative[1:])  
+                               )[0][0]
+            x_offset = ( 2*int(np.ceil( 
+                           0.5* (row_nPMT.max()  - row_nPMT[n_row+1]))) 
+                         - (row_nPMT[n_row+1]+1)%2)
+            i_PMT = i - row_nPMT_cumulative[n_row]
+            x_coord = x_offset + 2*i_PMT 
+            y_coord = tot_rows - n_row
+            coords[i] = [x_coord, y_coord]
+            pairs.append( (x_coord, y_coord))
+        return(coords,np.array(pairs) )
+    
+    def __init__(self):
+        self.coords,self.pairs = self.PMTnumuber_doublewidth()
+        self.size = (33,19)
+        
+    def get_coordinates(self):
+        """
+        this function returns dictionary of cooridnates
+        if form dict(ch: [x,y]) 
+        """
+        return(deepcopy(self.coords))
+    
+    def get_size(self):
+        """ 
+        return size of the 2D image 
+        """
+        return(deepcopy(self.size))
+    
+    def convert_pattern(self, inarr):
+        """
+        This function converts an array of PMT areas to 2D pattern
+        """
+        pattern = np.zeros(self.size)
+        pattern[self.pairs[:,0],self.pairs[:,1] ] = np.array(inarr)
+        return(pattern)
+    
+    def convert_multiple_patterns(self, inarr):
+        """
+        This function converts a 2D array of PMT arreas (for multiple events)
+        """
+        pattern = np.zeros((inarr.shape[0],self.size[0],self.size[1]))
+        pattern[:,self.pairs[:,0],self.pairs[:,1] ] = np.array(inarr)
+        return(pattern)
+
+if straxfound:
+    export, __all__ = strax.exporter()
+    @export
+    @strax.takes_config(
+        strax.Option('min_reconstruction_area',
+                     help='Skip reconstruction if area (PE) is less than this',
+                     default=10),
+        strax.Option('n_top_pmts', default=straxen.n_top_pmts,
+                     help="Number of top PMTs"),
+        strax.Option("s2_cnn_model_path", 
+                     help="Path to the CNN model in hdf5 format. WARING, this should include the whole model file and not the weights file", 
+                     default=("/project2/lgrandi/terliuk/CNNmodels/CNN_defaults/disk_const_PE/CNN_dw_3L_cm_maxnorm_A_const_100__Z_const_-1.hdf5")
```

Comment:
Please open a PR to https://github.com/XENONnT/private_nt_aux_files such that we can store this file in the database.

Reply:
Yes, i will add it there. It is about 800 KB, so I don't know if github is the best place to store such files.

Reply:
I made PR and it was merged already : https://github.com/XENONnT/private_nt_aux_files/pull/7